apps/api/src/users/users.service.ts

```diff
@@ -0,0 +1,36 @@
+import {Injectable} from '@nestjs/common';
+import {PrismaService} from '../prisma.service';
+import {UserDto} from '../auth/dto/user.dto';
+import {AuthMapper} from '../auth/auth.mapper';
+
+@Injectable()
+export class UsersService {
+  constructor(private prisma: PrismaService) {}
+
+  async allByRole(roleName: string): Promise<UserDto[]> {
+    const users = await this.prisma.user.findMany({
+      where: {
+        userRoles: {
+          some: {
+            role: {
+              name: roleName,
+            },
+          },
+        },
+      },
+      include: {
+        userRoles: {
+          include: {
+            role: {
+              include: {
+                permissions: true,
+              },
+            },
+          },
+        },
+      },
+    });
```

Comment:
This fetches the users along with their roles and permissions. It's overkill for this, but allowed for re-use of the auth mapper, so that's why I chose to go with it for now.